encoding/codecv2/codecv2.go

```diff
@@ -318,6 +326,13 @@ func EstimateBatchL1CommitBatchSizeAndBlobSize(b *encoding.Batch) (uint64, uint6
 	if err != nil {
 		return 0, 0, err
 	}
+	// Only apply this check when the uncompressed batch data has exceeded 128 KiB.
+	if len(batchBytes) > 131072 {
+		// Check compressed data compatibility.
+		if err = encoding.CheckCompressedDataCompatibility(blobBytes); err != nil {
```

Comment:
Let's add logs + metrics to these 2 error cases so that we can see how often they happen.

Reply:
We can also add these on the caller side (in `chunk-proposer` and `batch-proposer`), but in that case it's better to return an error here. Your call @colinlyguo

Reply:
add a new error type (`CompressedDataCompatibilityError`) for `chunk-proposer` and `batch-proposer` to catch, will submit metrics in `chunk-proposer` and `batch-proposer`.